.github/workflows/main.yml

```
@@ -27,6 +30,53 @@
+          defaults write org.x.X11 enable_test_extensions -boolean true
+          defaults write org.x.X11 app_to_run /opt/X11/bin/xterm
+          defaults write org.x.X11 no_auth -boolean true
+          defaults write org.x.X11 nolisten_tcp -boolean false
+          
+          # Create dummy display
+          mkdir -p /tmp/.X11-unix
+          sudo chmod 1777 /tmp/.X11-unix
+          sudo chown root /tmp/.X11-unix/
+
+      - name: Start X server and window
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2009:info:14:1: Consider using pgrep instead of grepping ps output

(shellcheck)